Library/Homebrew/cask/artifact/moved.rb

```diff
@@ -79,7 +94,18 @@ def move(adopt: false, force: false, verbose: false, command: nil, **options)
           end
         end
 
-        if target.dirname.writable?
+        if target.directory?
+          if target.writable?
+            source.children.each { |child| FileUtils.move(child, target + child.basename) }
+          else
+            command.run!("/bin/cp", args: ["-pR", "#{source}/*", "#{source}/.*", "#{target}/"],
+                                    sudo: true)
+          end
+          unless Quarantine.copy_xattrs(source, target)
```

Comment:
Looking at this again, I realise that I'm not actually sure what this is supposed to be doing. Is this only relevant for the `target.writable?` case when using ´FileUtils.move`?

Reply:
It's supposed to copy extended attributes for the topmost directory that's being copied. It is relevant for both cases; the "copy each child" will copy files and directories within the base directory and extended attributes on those files and directories, but not on the base directory itself.

Reply:
Okay, so if this is only needed for the topmost directory, do we need this huge Swift script?

Reply:
Let's try to avoid parsing hex dumps please.